File: InvoiceApp.Api/Models/PurchaseOrder.cs
namespace InvoiceApp.Api.Models;

public class PurchaseOrder
{
    public int Id { get; set; }
    public string Supplier { get; set; } = string.Empty;
    public List<OrderItem> Items { get; set; } = new();
    public string Status { get; set; } = "Pending";
}

public class OrderItem
{
    public int Id { get; set; }
    public int InventoryItemId { get; set; }
    public int Quantity { get; set; }
}

File: invoice_application/purchase_order.py
class PurchaseOrder:
    def __init__(self):
        self.orders = {}
        self.next_order_id = 1
    
    def create_order(self, supplier, items, quantity):
        """Create new purchase order"""
        order_id = f"PO-{self.next_order_id}"
        self.orders[order_id] = {
            'supplier': supplier,
            'items': items,
            'status': 'pending'
        }
        self.next_order_id += 1
        return order_id
    
    def update_order_status(self, order_id, status):
        """Update order status (pending/fulfilled/cancelled)"""
        if order_id in self.orders:
            self.orders[order_id]['status'] = status
    
    def get_orders(self):
        """Return all purchase orders"""
        return self.orders
    
